bin/git-delete-submodule

```diff
@@ -1,24 +1,49 @@
 #!/usr/bin/env bash
 
-test -z "$1" && echo "submodule required" 1>&2 && exit 1
-cd "$(git root)"
-test ! -f .gitmodules && echo ".gitmodules file not found" 1>&2 && exit 2
-
-NAME="$(echo "$1" | sed 's/\/$//g')"
-test -z \
-    "$(git config --file=.gitmodules submodule."$NAME".url)" \
-    && echo "submodule not found" 1>&2 && exit 3
-
-# 1. Delete the relevant section from .git/config and clean submodule files
-git submodule deinit -f "$NAME" || exit 4
-rmdir "$NAME"
-rm -rf .git/modules/"$NAME"
-# 2. Delete the relevant line from .gitmodules
-git config --file=.gitmodules --remove-section submodule."$NAME"
-git add .gitmodules
-# 3. Run git rm --cached path_to_submodule
-git rm --cached -rf "$NAME"
-# 4. Need to confirm and commit the changes for yourself
+abort() {
+	error="$1" && shift
+	echo "ERROR: $*" 1>&2
+	test -z "$FORCE" && exit "$error"
+}
+
+# Don't abort on failures.  This allows to cleanup after a previous failure.
+[ "$1" = '--force' ] && FORCE=1 && shift
+
+test -z "$1"            && abort 1 'Submodule required'
+cd "$(git root)"        || abort 5 'Cannot change to repository root'
+test ! -f '.gitmodules' && abort 2 '.gitmodules file not found'
+
+NAME="${1%/}"
+test -z "$(git config --file='.gitmodules' "submodule.$NAME.url")" \
+	   && abort 3 'Submodule not found'
+
+# 1. Handle the .git directory
+# 1.a. Delete the relevant section from .git/config
+git submodule deinit -f "$NAME" || abort 4 "Failed to deinitialize $NAME"
+# 1.b. Delete the submodule .git directory
+rm -rf ".git/modules/$NAME"
+# 1.c. Delete empty submodule directory
+git rm -f "$NAME"
+
+# 2. Handle .gitignore file
+# 2.a. Delete the relevant line from .gitmodules
+git config --file='.gitmodules' --remove-section "submodule.$NAME"
```

Comment:
It looks like the `submodule.$NAME` section is already removed by the `git rm -f` above. Running this command gets `fatal: No such section!` error. And the `.gitmodules` is added to the index too.
Maybe we just need to remove the empty `.gitmodules`?

Reply:
maybe this change is for later git versions?
I've ignored the error, but can change the code to remove it.
Please let me know.

Reply:
@go2null 
You are right. I tested it with an old version of git (1.8.3.1), and `git rm -f` didn't remove the section. A newer version of git (2.17) does.

Reply:
ok, so ignoring the error is good